### PR TITLE
Propagate and document timeout option (fixes #84)

### LINKED ISSUE #84: Document timeout option
The HTTP class supports a `timeout` option. We should document how to leverage it :) (and maybe its limitation) 

See https://github.com/Kinto/kinto.js/issues/280

### PR DESCRIPTION
**Pull request by leplatrem**
@n1k0 r?

### CODE REVIEW

File: src/base.js [hunk 1]
```diff
@@ -86,12 +87,13 @@ export default class KintoClientBase {
...
     this.events = options.events;
 
+    const {requestMode, timeout} = options;
     /**
      * The HTTP instance.
      * @ignore
      * @type {HTTP}
      */
-    this.http = new HTTP(this.events, {requestMode: options.requestMode});
+    this.http = new HTTP(this.events, {requestMode, timeout});
```
Comment (Natim) [inline src/base.js]: Nice cleanup :+1:

**Comment by n1k0** [comment]
r+

File: test/api_test.js [hunk 2]
```diff
@@ -79,6 +79,16 @@ describe("KintoClient", () => {
         .eql(requestMode);
     });
 
+    it("should keep the default timeout in the child HTTP instance", () => {
+      expect(new KintoClient(sampleRemote).http.timeout).eql(5000);
```
Comment (Natim) [inline test/api_test.js]: A default timeout of 5s seems a little small to me, usually it is more something around 30s isn't it?

Comment (leplatrem) [inline test/api_test.js]: I left the existing value. 5 seconds is already quite a lot I think, no?

Comment (Natim) [inline test/api_test.js]: This PR is about the documentation so we can discuss that in another one if we think 5 is not enough.

**Pull request merged**